server/portal/views/base.py

```diff
@@ -27,52 +26,62 @@ def dispatch(self, request, *args, **kwargs):
         """
         try:
             return super(BaseApiView, self).dispatch(request, *args, **kwargs)
-        except ApiException as err:
-
-            status = err.response.status_code
-            message = err.response.reason
-            extra = err.extra
+        except (PermissionDenied, Http404) as e:
+            # log information but re-raise exception to let django handle response
+            logger.error(e, exc_info=True)
+            raise e
+        except ApiException as e:
+            status = e.response.status_code
+            message = e.response.reason
+            extra = e.extra
             if status != 404:
                 logger.error(
                     '%s: %s',
                     message,
-                    err.response.text,
+                    e.response.text,
                     exc_info=True,
                     extra=extra
                 )
             else:
                 logger.info('Error %s', message, exc_info=True, extra=extra)
             return JsonResponse({'message': message}, status=400)
-        except (ConnectionError, HTTPError) as err:
-            status = err.response.status_code
-            content = err.response.json()
-            message = content.get("message", "Unknown Error")
-            if status in [404, 403]:
-                logger.warning(
-                    '%s: %s',
-                    message,
-                    err.response.text,
-                    exc_info=True,
-                    extra={
-                        'username': request.user.username,
-                        'session_key': request.session.session_key
-                    }
-                )
+        except (ConnectionError, HTTPError) as e:
+            if e.response is not None:
```

Comment:
If you remove this `if` block and pass an empty response, will that exception get caught by the clause in line 83? If that works it might be a way to remove an indentation level.

Reply:
It wouldn't be caught by line 83 as its part of the same  try/except block.